spec/suites/layer/GeoJSONSpec.js

```diff
@@ -512,3 +512,364 @@ describe("L.LayerGroup#toGeoJSON", function () {
 		});
 	});
 });
+
+describe("L.GeoJSON functions", function () {
+	describe("#geometryToLayer", function () {
+		var point = {
+			type: "Point",
+			coordinates: [0, 0]
+		};
+		var multiPoint  = {
+			type: "MultiPoint",
+			coordinates: [
+				[0, 0], [10, 10]
+			]
+		};
+		var line =  {
+			type: "LineString",
+			coordinates: [
+				[0, 0], [10, 10], [20, 20]
+			]
+		};
+		var multiLine = {
+			type: "MultiLineString",
+			coordinates: [
+				[[10, 10], [20, 20], [30, 30]],
+				[[50, 50], [60, 60], [70, 70]]
+			]
+		};
+		var polygon = {
+			type: "Polygon",
+			coordinates: [
+				[[30, 10], [40, 40], [20, 40], [10, 20], [30, 10]]
+			]
+		};
+		var multiPolygon = {
+			type: "MultiPolygon",
+			coordinates: [
+				[
+					[[30, 20], [45, 40], [10, 40], [30, 20]]
+				],
+				[
+					[[15, 5], [40, 10], [10, 20], [5, 10], [15, 5]]
+				]
+			]
+		};
+		var geometryCollection  = {
+			type: "GeometryCollection",
+			geometries: [
+				{
+					type: "Point",
+					coordinates: [0, 0]
+				},
+				{
+					type: "LineString",
+					coordinates: [
+						[10, 10], [20, 20]
+					]
+				}
+			]
+		};
+
+		function customPointToLayer(geojsonPoint, latLng) {
+			return L.circle(latLng, {
+				radius: geojsonPoint.properties.radius
+			});
+		}
+
+		function customCoordstoLatLng(coords) {
+			return L.latLng(coords[1] + 1, coords[0] + 1, coords[2] + 1);
+		}
+
+		[
+			point, line, polygon,
+			multiPoint, multiLine, multiPolygon,
+			geometryCollection
+		].forEach(function (geometry) {
+			it("creates a Layer from a GeoJSON feature (type='" + geometry.type + "')", function () {
+				var layer = L.GeoJSON.geometryToLayer({
+					type: "Feature",
+					geometry: geometry
+				});
+				expect(layer).to.be.a(L.Layer);
```

Comment:
I explicitly check for L.Layer because the documentation states that this function returns a L.Layer

Reply:
> and the docs reflect that in the way that it "only" returns a `L.Layer`.

The docs are brief, that is true, and many things are implied.
Atm every GeoJSON geometry has unambiguous Leaflet's match, except Point (that's why we have `pointToLayer` option).
As for the tests: there is no harm if they are slightly stricter that docs, as everyone can adapt them in case of need.
But believe me: you will want to know if someone accidently change returned layer type.

Reply:
Done with b57c7da